// @angular/fire/esm2015/storage/pipes/storageUrl.pipe.js
import { AsyncPipe } from '@angular/common';
import { ChangeDetectorRef, NgModule, Pipe } from '@angular/core';
import { AngularFireStorage } from '../storage';
/** to be used with in combination with | async */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../storage';
export class GetDownloadURLPipe {
    constructor(storage, cdr) {
        this.storage = storage;
        this.asyncPipe = new AsyncPipe(cdr);
    }
    transform(path) {
        if (path !== this.path) {
            this.path = path;
            this.downloadUrl$ = this.storage.ref(path).getDownloadURL();
        }
        return this.asyncPipe.transform(this.downloadUrl$);
    }
    ngOnDestroy() {
        this.asyncPipe.ngOnDestroy();
    }
}
GetDownloadURLPipe.ɵfac = function GetDownloadURLPipe_Factory(t) { return new (t || GetDownloadURLPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.AngularFireStorage), ɵngcc0.ɵɵinjectPipeChangeDetectorRef()); };
GetDownloadURLPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "getDownloadURL", type: GetDownloadURLPipe, pure: false });
/** @nocollapse */
GetDownloadURLPipe.ctorParameters = () => [
    { type: AngularFireStorage },
    { type: ChangeDetectorRef }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GetDownloadURLPipe, [{
        type: Pipe,
        args: [{
                name: 'getDownloadURL',
                pure: false
            }]
    }], function () { return [{ type: ɵngcc1.AngularFireStorage }, { type: ɵngcc0.ChangeDetectorRef }]; }, null); })();
export class GetDownloadURLPipeModule {
}
GetDownloadURLPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: GetDownloadURLPipeModule });
GetDownloadURLPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function GetDownloadURLPipeModule_Factory(t) { return new (t || GetDownloadURLPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(GetDownloadURLPipeModule, { declarations: [GetDownloadURLPipe], exports: [GetDownloadURLPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GetDownloadURLPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [GetDownloadURLPipe],
                exports: [GetDownloadURLPipe]
            }]
    }], null, null); })();
